contrib/lkt/language/parser.py

```diff
@@ -2208,6 +2208,41 @@ def expected_type_predicate(expected_type=T.TypeDecl.entity):
     def invalid_expected_type_error_name():
         return S("an array")
 
+    @langkit_property()
+    def expr_context_free_type():
+        """
+        Infer type from array elements context_free_type.
```

Comment:
Try to infer the array type from elements's context-free types*

(type inference is just tentative, and let’s use plain English when possible for the high-level explanation in the docstring)